lib/Command/Room/TRoomCommand.php

```diff
@@ -0,0 +1,222 @@
+	/**
+	 * @param Room $room
+	 * @param bool $readOnly
+	 *
+	 * @throws InvalidArgumentException
+	 */
+	protected function setRoomReadOnly(Room $room, bool $readOnly): void {
+		if ($readOnly === ($room->getReadOnly() === Room::READ_ONLY)) {
+			return;
+		}
+
+		if (!$room->setReadOnly($readOnly ? Room::READ_ONLY : Room::READ_WRITE)) {
+			throw new InvalidArgumentException('Unable to change room state.');
+		}
+	}
+
+	/**
+	 * @param Room   $room
+	 * @param string $password
+	 *
+	 * @throws InvalidArgumentException
+	 */
+	protected function setRoomPassword(Room $room, string $password): void {
+		if ($room->hasPassword() ? $room->verifyPassword($password)['result'] : ($password === '')) {
+			return;
+		}
+
+		if (($password !== '') && ($room->getType() !== Room::PUBLIC_CALL)) {
+			throw new InvalidArgumentException('Unable to add password protection to private room.');
+		}
+
+		if (!$room->setPassword($password)) {
+			throw new InvalidArgumentException('Unable to change room password.');
+		}
+	}
+
+	/**
+	 * @param Room     $room
+	 * @param string[] $userIds
+	 *
+	 * @throws InvalidArgumentException
+	 */
+	protected function addRoomParticipants(Room $room, array $userIds): void {
+		$userManager = \OC::$server->getUserManager();
+
+		$participants = [];
+		foreach ($userIds as $userId) {
+			$user = $userManager->get($userId);
+			if ($user === null) {
+				throw new InvalidArgumentException(sprintf("User '%s' not found.", $userId));
+			}
+
+			try {
+				$room->getParticipant($user->getUID());
+				// nothing to do, user already is a participant
+			} catch (ParticipantNotFoundException $e) {
+				$participants[] = [
+					'userId' => $user->getUID(),
+				];
+			}
+		}
+
+		\call_user_func_array([$room, 'addUsers'], $participants);
+	}
+
+	/**
+	 * @param Room     $room
+	 * @param string[] $userIds
+	 *
+	 * @throws InvalidArgumentException
+	 */
+	protected function removeRoomParticipants(Room $room, array $userIds): void {
+		$userManager = \OC::$server->getUserManager();
+
+		$users = [];
+		foreach ($userIds as $userId) {
+			try {
+				$room->getParticipant($userId);
+			} catch (ParticipantNotFoundException $e) {
+				throw new InvalidArgumentException(sprintf("User '%s' is no participant.", $userId));
+			}
+
+			$users[] = $userManager->get($userId);
+		}
+
+		foreach ($users as $user) {
+			$room->removeUser($user, Room::PARTICIPANT_REMOVED);
+		}
+	}
+
+	/**
+	 * @param Room     $room
+	 * @param string[] $userIds
+	 *
+	 * @throws InvalidArgumentException
+	 */
+	protected function addRoomModerators(Room $room, array $userIds): void {
+		$participants = [];
+		foreach ($userIds as $userId) {
+			try {
+				$participants[] = $room->getParticipant($userId);
+			} catch (ParticipantNotFoundException $e) {
+				throw new InvalidArgumentException(sprintf("User '%s' is no participant.", $userId));
+			}
+		}
+
+		foreach ($participants as $participant) {
+			$room->setParticipantType($participant, Participant::MODERATOR);
```

Comment:
You need to check that the user was a normal user before. It should not allow "demoting" owners to moderators